app/services/association_authorizer.rb

```diff
@@ -15,6 +15,7 @@ def self.authorized_associations(associations,  klass_name = nil, should_raise_e
 
   def self.permission_name(klass, permission, should_raise_exception)
     klass = klass.first if klass.is_a?(Array)
+    klass = klass.class unless klass.is_a?(Class)
```

Comment:
I think the patch belongs in the previous method - `authorized_associations`. `associations_klass` might end up being just 'associations' ([casbios] in your case). Can you add a test for this behavior?